charts/templates/statefulset.yaml

```diff
@@ -174,12 +174,28 @@ spec:
                         log "[poststart] wait for marklogic server to be ready"
                         sleep 5s
                       done
-                      sleep 10s
+
+                      # Fetch timestamp before updating group config
+                      TIMESTAMP=`curl --anyauth http://${MARKLOGIC_BOOTSTRAP_HOST}:8001/admin/v1/timestamp --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}`
+
                       GROUP_CFG_TEMPLATE='{"group-name":"%s", "xdqp-ssl-enabled":"%s"}'
                       GROUP_CFG=$(printf "$GROUP_CFG_TEMPLATE" "$MARKLOGIC_GROUP" "$XDQP_SSL_ENABLED") 
                       log "Info: [poststart] Updating group configuration: ${GROUP_CFG}"
-                      curl --anyauth -m 20 -X PUT -H "Content-type: application/json" -d "${GROUP_CFG}" http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups/Default/properties --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}
-                      sleep 2s
+                      GROUP_RESP_CODE=`curl --retry 5 --retry-max-time 60 -o /dev/null -w "%{http_code}" --anyauth -m 20 -s -X PUT -H "Content-type: application/json" -d "${GROUP_CFG}" http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups/Default/properties --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}`
+                      
+                      # check if MarkLogic has restarted
```

Comment:
According to https://docs.marklogic.com/REST/PUT/manage/v2/groups/[id-or-name]/properties, we should also check for 204 response code, and if it is not 204 means the update group failed and it should log the error message.

Reply:
I do agree with Peng comment here we should check both 202 and 204.

Reply:
I agree with adding a check but for bug CLD-773, it was required to remove sleep statements. 
So the resolution was to add retry for the curl call. And the check for response code 202 is added in order to wait until MarkLogic has restarted before completing the post-start hook execution when any group setting is updated that causes a restart. This check wasn't added to make sure the group is updated successfully.
The existing code also has no check for success/error. So I think we can add the check as a different spike for bug.

Reply:
Added check for 204